projects/NodeBB/services/nodebb/examples/postgresql.nix

```diff
@@ -0,0 +1,32 @@
+{ pkgs, ... }:
+
+{
+  services.nodebb = {
+    enable = true;
+    admin = {
+      username = "admin";
+      email = "admin@example.com";
+      passwordFile = pkgs.writeText "nodebb-admin-password" "nodebb";
+    };
+    database = {
+      type = "postgres";
+      passwordFile = pkgs.writeText "postgresql-password" "nodebb";
+    };
+  };
+
+  systemd.services.nodebb.after = [ "postgresql.service" ];
```

Comment:
We can set this in the nodebb module if the `database.type` is "postgres". Same for the `services.postgresql` config.

Reply:
Users may use PostgreSQL from different sources (e.g. Docker) so we would want the user to decide what to use here.